lib/discord.go

```diff
@@ -242,6 +385,63 @@ func doDiscordReq(ctx context.Context, path string, method string, body io.ReadC
 
 		RequestHistogram.With(map[string]string{"route": route, "status": status, "method": method, "clientId": identifier.(string)}).Observe(elapsed)
 	}
+
+	if wsProxy != "" && discordResp.StatusCode == 200 {
+		var isGwProxyUrl bool
+
+		if path == "/api/gateway" || path == "/api/gateway/bot" {
+			isGwProxyUrl = true
+		} else if ok, _ := filepath.Match("/api/v*/gateway/bot", path); ok {
+			isGwProxyUrl = true
+		} else if ok, _ := filepath.Match("/api/v*/gateway", path); ok {
+			isGwProxyUrl = true
+		}
```

Comment:
This can be simplified a lot by checking the suffixes (its also more performant):

```suggestion
		if strings.HasSuffix(path, "/gateway") || strings.HasSuffix(path, "/gateway/bot") {  
			isGwProxyUrl = true
		}
```